### PR TITLE
add review test report tutorial

### PR DESCRIPTION
**Pull request by nancyc12**
## Description

Add "Review Test Report" section to the base tutorial.

## Resolved issues

[CHECKBOX-758](https://warthogs.atlassian.net/browse/CHECKBOX-758)

## Documentation

Documentation itself.

## Tests

Tested locally by running `make install` and `make run`.
Also ran `make woke`, `make spelling` and `make linkcheck`. 
`make spelling` failed with "JUnit", which is expected.

[CHECKBOX-758]: https://warthogs.atlassian.net/browse/CHECKBOX-758?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/tutorial/using-checkbox/test-report.rst [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+    ``☒`` job failed
+
+    ``☐`` job skipped, job cannot be started
+
+    ``‒`` job is not implemented
+
+    ``⁇`` job needs verification
+
+    ``⚠`` job crashed
```
Comment (Hook25) [inline docs/tutorial/using-checkbox/test-report.rst:40]: This renders a bit weird due to tabs, consider changing this to a list

File: docs/tutorial/using-checkbox/test-report.rst [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+.. _submission-files:
+Submission files
+------------
```
Comment (Hook25) [inline docs/tutorial/using-checkbox/test-report.rst:44]: Minor: make complains, too little dashes

File: docs/tutorial/using-checkbox/test-report.rst [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+    ``☐`` job skipped, job cannot be started
+
+    ``‒`` job is not implemented
+
+    ``⁇`` job needs verification
+
+    ``⚠`` job crashed
+
+.. _submission-files:
```
Comment (Hook25) [inline docs/tutorial/using-checkbox/test-report.rst:42]: Minor: make complains, no newline after marker

File: docs/tutorial/using-checkbox/test-report.rst [hunk 4]
```diff
@@ -0,0 +1,87 @@
+.. _test-report:
+
+Review Test Report 
+======================
+
+Checkbox optionally generates test reports in different formats that can be used to easily share the results of a test session. Generally, a :ref:`text summary<test-summary>` and :ref:`submission files<submission-files>` would be created. In this section, you will learn where to find the report files and what they contain.
+
+Note that you can tailor desired reports in your launcher file, or define your own exporter to customize the reports. See :doc:`launcher<../../reference/launcher>` and :doc:`exporter<../../reference/units/exporter>` for more details.
+
+.. _test-summary:
```
Comment (Hook25) [inline docs/tutorial/using-checkbox/test-report.rst:10]: Minor: make complains, no new line after marker

File: docs/tutorial/using-checkbox/test-report.rst [hunk 5]
```diff
@@ -0,0 +1,87 @@
+.. _test-report:
+
+Review Test Report 
+======================
```
Comment (Hook25) [inline docs/tutorial/using-checkbox/test-report.rst:4]: Minor: make complains, too many equals

File: docs/tutorial/using-checkbox/test-report.rst [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+Checkbox optionally generates test reports in different formats that can be used to easily share the results of a test session. Generally, a :ref:`text summary<test-summary>` and :ref:`submission files<submission-files>` would be created. In this section, you will learn where to find the report files and what they contain.
```
Comment (Hook25) [inline docs/tutorial/using-checkbox/test-report.rst:6]: I would rephrase this as: Checkbox generates test reports [...]. By default, a test-summary and the submission files are created.

File: docs/tutorial/using-checkbox/test-report.rst [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+Note that you can tailor desired reports in your launcher file, or define your own exporter to customize the reports. See :doc:`launcher<../../reference/launcher>` and :doc:`exporter<../../reference/units/exporter>` for more details.
```
Comment (Hook25) [inline docs/tutorial/using-checkbox/test-report.rst:8]: you can change this to an actual note
```
.. note::

  You can tailor desired reports in your launcher file, or define your own exporter to customize the reports. See :doc:`launcher<../../reference/launcher>` and :doc:`exporter<../../reference/units/exporter>` for more details.
```

Nice suggestion though!

File: docs/tutorial/using-checkbox/test-report.rst [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+
+.. _test-summary:
+Text summary
+------------
+
+A text summary shows in the console once all jobs complete, providing an overview of test result. You can find the result of each job displaying in a form of ``outcome: summary``.
```
Comment (Hook25) [inline docs/tutorial/using-checkbox/test-report.rst:14]: I would rephrase this as: A text summary is displayed in the console once all jobs are completed. Each result is in the form: ``outcome: summary``.

File: docs/tutorial/using-checkbox/test-report.rst [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+
+    ``⁇`` job needs verification
+
+    ``⚠`` job crashed
+
+.. _submission-files:
+Submission files
+------------
+
+In Checkbox, the submission files are used for sharing test results to Jenkins and Certification Website. Submission files contain the following files:
```
Comment (Hook25) [inline docs/tutorial/using-checkbox/test-report.rst:46]: I would rephrase this to speak more about the fact that these files (and the multiple formats) are useful to share the results with services or other users. 

Minor: first mention of the certification website, maybe a link could be useful here?

File: docs/tutorial/using-checkbox/test-report.rst [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+    file:///home/user/.local/share/checkbox-ng/submission.html
+    file:///home/user/.local/share/checkbox-ng/submission.junit.xml
+    file:///home/user/.local/share/checkbox-ng/submission.tar.xz
+
+``html``
+    Self-contained HTML files contain the following sections.
+
+        - System Information
+        - Tests Results 
+        - Logs
```
Comment (Hook25) [inline docs/tutorial/using-checkbox/test-report.rst:74]: this renders a bit weird due to tabs, consider aligning with Self

File: docs/tutorial/using-checkbox/test-report.rst [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+
+.. figure:: ../../_images/checkbox-test-report.png
+    
+    An example of beginning of a HTML report
+
+``json``
+    JSON files contain session export compatible for submission to Certification Website.
+
+``junit``
+    JUnit XML files contain test data that can be read by Jenkins.
```
Comment (Hook25) [inline docs/tutorial/using-checkbox/test-report.rst:84]: Minor: link to the junit website may be useful here

File: docs/tutorial/using-checkbox/test-report.rst [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+        - System Information
+        - Tests Results 
+        - Logs
+
+.. figure:: ../../_images/checkbox-test-report.png
+    
+    An example of beginning of a HTML report
+
+``json``
+    JSON files contain session export compatible for submission to Certification Website.
```
Comment (Hook25) [inline docs/tutorial/using-checkbox/test-report.rst:81]: Minor: I dont know if we have a spec for this, but if we do we may want to link it here (like a reference of some sort)

**Review by Hook25 — CHANGES_REQUESTED** [review]
Very nice job overall, here are a few small changes I would make

Not included below but I think that we should go to a new line around 80 characters in to make this more readable from a text editor and make it easier to comment on it in the future

File: docs/tutorial/using-checkbox/test-report.rst (hunk 6 above)
Comment (pieqq) [reply docs/tutorial/using-checkbox/test-report.rst:6]: "text summary" is the right wording, maybe the reference can be renamed to `text-summary` instead :)

Otherwise, I agree with Max, let's talk about the default behavior of Checkbox (the customized behavior is discussed about in the launcher section anyway).

File: docs/tutorial/using-checkbox/test-report.rst (hunk 9 above)
Comment (pieqq) [reply docs/tutorial/using-checkbox/test-report.rst:46]: Just for info, we have a glossary with [an entry for the Cert Website](https://checkbox.readthedocs.io/en/latest/reference/glossary.html#term-Certification-Website), and you can point to it in your doc with

```
:term:`Certification website`
```

File: docs/tutorial/using-checkbox/test-report.rst [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+
+``junit``
+    JUnit XML files contain test data that can be read by Jenkins.
+
+``tar``
```
Comment (pieqq) [inline docs/tutorial/using-checkbox/test-report.rst:86]: `tar.xz`, since it's a tarball archive compressed with lzma.

File: docs/tutorial/using-checkbox/test-report.rst [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+    Xz compressed tarball of the HTML, JUnit and JSON reports. Also contains all the attachments (I/O logs and binary files). Certification Website only accepts submissions tarballs, from which it extracts the submission.json file to create a new test report in the database.
```
Comment (pieqq) [inline docs/tutorial/using-checkbox/test-report.rst:87]: I would put submission.json as `submission.json` (`<code>` in HTML, done with two ticks `` in reStructuredText).

**Review by pieqq — CHANGES_REQUESTED** [review]
Thanks for this doc! I added a few comments/suggestions inline.

I don't know if you've read Diataxis page about Tutorials: https://diataxis.fr/tutorials.html

One of the key in writing a tutorial is that you, as the tutorial author, have to take the user by the hand, tell them what to do, what to observe, and how to achieve something.

We want the user to be running a sample tutorial test plan/test cases (but it's not ready yet, see [CHECKBOX-750](https://warthogs.atlassian.net/browse/CHECKBOX-750)), and once the user has gone through this, the text summary and the submission files will be available. Then you can ask them to review the text summary (for instance "check that <a given job that we expect in the outcome> is here"), or "open the HTML file located at ..., you should see this and that".

Not sure

File: docs/tutorial/using-checkbox/test-report.rst (hunk 9 above)
Comment (nancyc12) [reply docs/tutorial/using-checkbox/test-report.rst:46]: yes I also noticed that we have it in the glossary. Have made the same change =)

File: docs/tutorial/using-checkbox/test-report.rst (hunk 14 above)
Comment (nancyc12) [reply docs/tutorial/using-checkbox/test-report.rst]: fixed in [fdd5e20](https://github.com/canonical/checkbox/pull/637/commits/fdd5e208e18dfabfacf2c8bf416d1b017b73251b)

File: docs/tutorial/using-checkbox/test-report.rst (hunk 13 above)
Comment (nancyc12) [reply docs/tutorial/using-checkbox/test-report.rst]: fixed in [fdd5e20](https://github.com/canonical/checkbox/pull/637/commits/fdd5e208e18dfabfacf2c8bf416d1b017b73251b)

File: docs/tutorial/using-checkbox/test-report.rst (hunk 11 above)
Comment (nancyc12) [reply docs/tutorial/using-checkbox/test-report.rst]: fixed in [fdd5e20](https://github.com/canonical/checkbox/pull/637/commits/fdd5e208e18dfabfacf2c8bf416d1b017b73251b)

File: docs/tutorial/using-checkbox/test-report.rst (hunk 10 above)
Comment (nancyc12) [reply docs/tutorial/using-checkbox/test-report.rst]: fixed in [fdd5e20](https://github.com/canonical/checkbox/pull/637/commits/fdd5e208e18dfabfacf2c8bf416d1b017b73251b)

File: docs/tutorial/using-checkbox/test-report.rst (hunk 9 above)
Comment (nancyc12) [reply docs/tutorial/using-checkbox/test-report.rst]: fixed in [fdd5e20](https://github.com/canonical/checkbox/pull/637/commits/fdd5e208e18dfabfacf2c8bf416d1b017b73251b)

File: docs/tutorial/using-checkbox/test-report.rst (hunk 8 above)
Comment (nancyc12) [reply docs/tutorial/using-checkbox/test-report.rst]: fixed in [fdd5e20](https://github.com/canonical/checkbox/pull/637/commits/fdd5e208e18dfabfacf2c8bf416d1b017b73251b)

File: docs/tutorial/using-checkbox/test-report.rst (hunk 5 above)
Comment (nancyc12) [reply docs/tutorial/using-checkbox/test-report.rst]: fixed in [fdd5e20](https://github.com/canonical/checkbox/pull/637/commits/fdd5e208e18dfabfacf2c8bf416d1b017b73251b)

File: docs/tutorial/using-checkbox/test-report.rst (hunk 4 above)
Comment (nancyc12) [reply docs/tutorial/using-checkbox/test-report.rst]: fixed in [fdd5e20](https://github.com/canonical/checkbox/pull/637/commits/fdd5e208e18dfabfacf2c8bf416d1b017b73251b)

File: docs/tutorial/using-checkbox/test-report.rst (hunk 3 above)
Comment (nancyc12) [reply docs/tutorial/using-checkbox/test-report.rst]: fixed in [fdd5e20](https://github.com/canonical/checkbox/pull/637/commits/fdd5e208e18dfabfacf2c8bf416d1b017b73251b)

File: docs/tutorial/using-checkbox/test-report.rst (hunk 2 above)
Comment (nancyc12) [reply docs/tutorial/using-checkbox/test-report.rst]: fixed in [fdd5e20](https://github.com/canonical/checkbox/pull/637/commits/fdd5e208e18dfabfacf2c8bf416d1b017b73251b)

File: docs/tutorial/using-checkbox/test-report.rst (hunk 1 above)
Comment (nancyc12) [reply docs/tutorial/using-checkbox/test-report.rst]: fixed in [fdd5e20](https://github.com/canonical/checkbox/pull/637/commits/fdd5e208e18dfabfacf2c8bf416d1b017b73251b)

File: docs/tutorial/using-checkbox/test-report.rst (hunk 6 above)
Comment (nancyc12) [reply docs/tutorial/using-checkbox/test-report.rst]: fixed in [fdd5e20](https://github.com/canonical/checkbox/pull/637/commits/fdd5e208e18dfabfacf2c8bf416d1b017b73251b)

File: docs/tutorial/using-checkbox/test-report.rst (hunk 7 above)
Comment (nancyc12) [reply docs/tutorial/using-checkbox/test-report.rst]: fixed in [fdd5e20](https://github.com/canonical/checkbox/pull/637/commits/fdd5e208e18dfabfacf2c8bf416d1b017b73251b)

File: docs/tutorial/using-checkbox/test-report.rst (hunk 12 above)
Comment (nancyc12) [reply docs/tutorial/using-checkbox/test-report.rst]: According to Matias, there's no JSON schema for this currently. The reference link can be added in the future once we have the schema.

File: docs/tutorial/using-checkbox/test-report.rst [hunk 15]
```diff
@@ -0,0 +1,129 @@
+.. _test-report:
+
+Review Test Report 
+==================
+
+Once you finish your first test run, you may want to review the test reports. 
+Checkbox generates test reports in different formats that can be used to 
+easily share the results of a test session. By default, a :ref:`text summary<
```
Comment (Hook25) [inline docs/tutorial/using-checkbox/test-report.rst:8]: for some reason this doesn't work, put the link all on one line or it will not render (the `<` captures all chars, including the new line -> `WARNING: undefined label: ' text-summary'`)

File: docs/tutorial/using-checkbox/test-report.rst [hunk 16]
```diff
@@ -0,0 +1,129 @@
...
+text-summary>` and :ref:`submission files<submission-files>` are created. This 
+section will provide you with information on where to locate the report files 
+and their respective contents.
+
+.. note::
+
+    You can tailor desired reports in your launcher file, or define your own 
+    exporter to customize the reports. See :doc:`launcher<../../reference/
```
Comment (Hook25) [inline docs/tutorial/using-checkbox/test-report.rst:16]: this link, fix it the same way `WARNING: unknown document: '../../reference/ launcher'`

**Review by Hook25 — APPROVED** [review]
Content is on point, very nice job. Before merging fix this two links and also, ad JUnit to the wordlist!

File: docs/tutorial/using-checkbox/test-report.rst (hunk 15 above)
Comment (nancyc12) [reply docs/tutorial/using-checkbox/test-report.rst]: fixed in [6159a36](https://github.com/canonical/checkbox/pull/637/commits/6159a3674bc44d689c7a6ab7acda85d5f68d13d1).

File: docs/tutorial/using-checkbox/test-report.rst (hunk 16 above)
Comment (nancyc12) [reply docs/tutorial/using-checkbox/test-report.rst]: fixed in [6159a36](https://github.com/canonical/checkbox/pull/637/commits/6159a3674bc44d689c7a6ab7acda85d5f68d13d1)

**Review by pieqq — APPROVED** [review]
+1, thanks a lot!
